ldk/javascript/examples/self-test-loop/src/tests/filesystem/index.ts

```diff
@@ -35,128 +36,116 @@ export const queryDirectory = (): Promise<boolean> =>
         await filesystem.remove(`${dirPath}/file.json`);
       });
   });
 
-export const createAndDeleteFile = (): Promise<boolean> =>
+export const testWriteAndRemoveFile = (): Promise<boolean> =>
   new Promise(async (resolve, reject) => {
     const filePath = `${await getTestFolderPath()}/test.txt`;
     const writeMode = 0o755;
 
     network
       .encode('some text')
       .then((encodedValue) => {
-        setTimeout(() => {
-          filesystem
-            .writeFile({
-              path: filePath,
-              data: encodedValue,
-              writeOperation: filesystem.WriteOperation.overwrite,
-              writeMode,
-            })
-            .then(() => {
-              filesystem
-                .remove(filePath)
-                .then(() => {
-                  setTimeout(() => {
-                    resolve(true);
-                  }, 1500);
-                })
-                .catch((error) => {
-                  setTimeout(() => {
-                    reject(error);
-                  }, 1500);
-                });
-            })
-            .catch((error) => {
-              console.error('write file failed');
-              reject(error);
-            });
-        }, 500);
+        filesystem
+          .writeFile({
+            path: filePath,
+            data: encodedValue,
+            writeOperation: filesystem.WriteOperation.overwrite,
+            writeMode,
+          })
+          .then(() => {
+            filesystem
+              .remove(filePath)
+              .then(() => {
+                resolve(true);
+              })
+              .catch((error) => {
+                reject(error);
+              });
+          })
+          .catch((error) => {
+            console.error('write file failed');
+            reject(error);
+          });
       })
       .catch((error) => {
         reject(error);
       });
   });
 
-export const updateAndReadFile = (): Promise<boolean> =>
+export const testWriteAndReadFile = (): Promise<boolean> =>
   new Promise(async (resolve, reject) => {
     const testString = 'Im in yr loop, writing to yr clipboard';
     const filePath = `${await getTestFolderPath()}/test.txt`;
     const writeMode = 0o755;
 
-    setTimeout(() => {
-      network
-        .encode(testString)
-        .then((encodedValue) => {
-          filesystem
-            .writeFile({
-              path: filePath,
-              data: encodedValue,
-              writeOperation: filesystem.WriteOperation.overwrite,
-              writeMode,
-            })
-            .then(() => {
-              console.debug('Write successful');
-              console.debug(encodedValue);
+    network
+      .encode(testString)
+      .then((encodedValue) => {
+        filesystem
+          .writeFile({
+            path: filePath,
+            data: encodedValue,
+            writeOperation: filesystem.WriteOperation.overwrite,
+            writeMode,
+          })
+          .then(() => {
```

Comment:
i see you eradicated some of the `.then`s, but not others. do we have plans to remove them all?

Reply:
I could sweep through a whole thing. Initially I tried to keep changes to a min to reduce conflicts. But had to touch lots of things with timeout removal and took care of everything what I touched. We are close, perhaps we will improve while we go.